fenicsadapter/fenicsadapter.py

```diff
@@ -55,54 +64,98 @@ def eval(self, value, x):
 
 
 class Adapter(object):
+    """Initializes the Adapter. Initalizer creates object of class Config (from
+    config.py module).
+
+    :ivar _config: object of class Config, which stores data from config file
+    :ivar _solver_name: name of the solver (from JSON config file)
+    :ivar _interface: coupling interface
+    :ivar _dimensions: number of dimensions
+    :ivar _coupling_subdomain: FEniCS subdomain defining the coupling interface
+    :ivar _mesh_fenics: FEniCS mesh where the coupled simulation takes place
+    :ivar _coupling_bc_expression: expression describing the coupling bc condition
+    :ivar _coupling_mesh_vertices: mesh vertices in a format that can be understood by preCICE
+    :ivar _mesh_name: name of mesh as defined in preCICE config (from JSON config file)
+    :ivar _mesh_id: ID of the coupling mesh created from mesh name
+    :ivar _vertex_ids: ID of vertices filled by preCICE
+    :ivar _n_vertices: number of vertices
+    :ivar _write_data_name: name of write data as defined in preCICE config (from JSON config file)
+    :ivar _write_data_id: ID of the data on the coupling mesh created from data name
+    :ivar _write_data: actual data to write
+    :ivar _read_data_name: name of read data as defined in preCICE config (from JSON config file)
+    :ivar _read_data_id: ID of the data on the coupling mesh created from data name
+    :ivar _read_data: actual data to read
+    :ivar _precice_tau: timestep
+
+    :func convert_fenics_to_precice(): converts FEniCS function, into an array of values
+    :func set_coupling_mesh(): sets the coupling mesh
+    :func extract_coupling_boundary_vertices(): extracts verticies which lay on the boundary
+    :func set_coupling_mesh: sets the coupling mesh
+    :func set_write_field: sets the write field
+    :func set_read_field: sets the read field
+    :func create_coupling_boundary_condition():
+    :func create_coupling_dirichlet_boundary_condition():
+    :func create_coupling_neumann_boundary_condition():
+    :func advance(): advances the simulation
+    :func initialize(): initializes remaining attributes
+    :func is_coupling_ongoing(): determines whether simulation should continue
+    :func extract_coupling_boundary_coordinates(): extracts the coordinates of boundary vertices
+    :func finalize(): finalizes the coupling interface.
+    """
     def __init__(self):
-        self._solver_name = None  # name of the solver, will be configured later
+        self._config = Config()
 
-        self._interface = None  # coupling interface, will be initialized later
+        self._solver_name = self._config._solver_name
 
-        self._dimensions = None
+        self._interface = PySolverInterface.PySolverInterface(self._solver_name, 0, 1)
+        self._interface.configure(self._config._config_file_name)
```

Comment:
`self._config._config_file_name` is by [convention](https://stackoverflow.com/a/2064400/5158031) an internal variable of `Config`. There are two possible solutions:

1. make it "public", i.e. remove the leading `_` of `_config_file_name`.
2. write an appropriate getter function. 

I think that the preferrable (though more verbose) option is number 2, since we do not want that `_config_file_name` is changed from the outside.

Reply:
Agree.